src/Decompiler/Evaluation/ExpressionSimplifier.cs

```diff
@@ -553,6 +559,25 @@ public virtual Expression VisitPhiFunction(PhiFunction pc)
             }
         }
 
+        private Expression SimplifyPhiArg(Expression arg)
```

Comment:
>Could you not use the VisitBinaryExpression method to perform the same simplification?

No, it could not simplify following statements:
```
y = x - 4
a = y + 4
x = phi(a, b)
```
to
```
y = x - 4
a = x
x = phi(a, b)
```
`IdBinIdc` class processes `y` as 'used in phi' and prevents propagation